src/components/loader-spinner/loader-spinner.style.ts

```diff
@@ -0,0 +1,176 @@
+import styled, { css } from "styled-components";
+import { margin } from "styled-system";
+import { LoaderSpinnerProps } from "./loader-spinner.component";
+import { LOADER_SPINNER_SIZE_PARAMS } from "./loader-spinner.config";
+import Typography from "../typography";
+
+const calculateColors = (WheelType: "Wheel" | "Wrapper", color?: string) => {
+  const wheelType = WheelType === "Wheel";
+
+  switch (color) {
+    case "neutral":
+      return wheelType
+        ? "var(--colorsSemanticNeutral500);"
+        : "var(--colorsSemanticNeutral200);";
+    case "gradient-grey":
+      return wheelType ? "#00D639" : "#0000001A";
+    case "gradient-white":
+      return wheelType ? "#00D639" : "var(--colorsActionMajorYang100);";
+    case "inverse":
+      return wheelType
+        ? "var(--colorsActionMajorYang100);"
+        : "rgba(255, 255, 255, 0.3)";
+    default:
+      return wheelType
+        ? "var(--colorsActionMajor500);"
+        : "var(--colorsActionMajor150);";
+  }
+};
+
+interface StyledSpinnerWrapperProps
+  extends Pick<LoaderSpinnerProps, "size" | "color"> {
+  reduceMotion?: boolean;
+}
+
+export const StyledSpinnerWrapper = styled.div<StyledSpinnerWrapperProps>`
+  ${margin}
+  ${({ size, reduceMotion, color }) =>
+    size &&
+    css`
+      width: ${LOADER_SPINNER_SIZE_PARAMS[size].wrapperDimensions}px;
+      height: ${LOADER_SPINNER_SIZE_PARAMS[size].wrapperDimensions}px;
+      min-width: ${LOADER_SPINNER_SIZE_PARAMS[size].wrapperDimensions}px;
+      min-height: ${LOADER_SPINNER_SIZE_PARAMS[size].wrapperDimensions}px;
+
+      ${!reduceMotion &&
+      css`
+        box-shadow: inset 0px 0px 0px
+          ${LOADER_SPINNER_SIZE_PARAMS[size].boxShadow}px
+          ${calculateColors("Wrapper", color)};
+        border-radius: 50%;
+      `}
+    `}
+`;
+
+export const StyledLabel = styled(Typography)`
+  display: flex;
+  justify-content: center;
+`;
+
+export const StyledSpinnerCircleSvg = styled.svg<
+  Omit<LoaderSpinnerProps, "showSpinnerLabel" | "labelColor">
+>`
+  ${({ size, tracked, motion, animationTime, color }) =>
+    size &&
+    css`
+      x: 0px;
+      y: 0px;
+
+      circle {
+        fill: transparent;
+        stroke-width: ${LOADER_SPINNER_SIZE_PARAMS[size].strokeWidth}px;
+        stroke: ${calculateColors("Wheel", color)}
+        stroke-linecap: round;
+
+        stroke-dasharray: 100px;
+        stroke-dashoffset: 80px;
+
+        transform-origin: 12px 12px 0px;
+
+        cx: 12px;
+        cy: 12px;
+        r: 10px;
```

Comment:
**suggestion**: making both the spinner circle and animated arc SVGs should resolve the issue of the arc not fully covering the spinner, as previously discussed externally: 

https://github.com/Sage/carbon/assets/18368713/b2370c89-e59d-4abc-a5ae-300da7f89371


```suggestion
      circle[data-element="outer-arc"] {
        fill: transparent;
        stroke-width: ${LOADER_SPINNER_SIZE_PARAMS[size].strokeWidth}px;
        stroke: ${calculateColors(false, color)};
        stroke-opacity: 0.5;
        cx: 12px;
        cy: 12px;
        r: 10px;
      }

      circle[data-element='spinner-circle'] {
        fill: transparent;
        stroke-width: ${LOADER_SPINNER_SIZE_PARAMS[size].strokeWidth + 0.1}px;
        stroke: ${calculateColors(true, color)}
        stroke-linecap: round;

        stroke-dasharray: 100px;
        stroke-dashoffset: 80px;

        transform-origin: 12px 12px 0px;
        cx: 12px;
        cy: 12px;
        r: 10px;
```

and in `loader-spinner.component.tsx`:

```tsx
<StyledSpinnerCircleSvg role="presentation" viewBox="0 0 24 24"> {...spinnerProps}>
  <circle data-element="outer-arc" />
  <circle data-element="spinner-circle" />
</StyledSpinnerCircleSvg>
```